hack/make/test-unit

```diff
@@ -25,15 +25,30 @@ bundle_test_unit() {
 	else
 		TEST_PATH=./${TESTDIRS}
 	fi
-	pkg_list=$(go list -e \
-		-f '{{if ne .Name "github.com/docker/docker"}}
-			{{.ImportPath}}
-		    {{end}}' \
-		"${BUILDFLAGS[@]}" $TEST_PATH \
-		| grep github.com/docker/docker \
-		| grep -v github.com/docker/docker/vendor \
-		| grep -v github.com/docker/docker/man \
-		| grep -v github.com/docker/docker/integration-cli)
+
+	if [ "$(go env GOHOSTOS)" = 'solaris' ]; then
+		pkg_list=$(go list -e \
+			-f '{{if ne .Name "github.com/docker/docker"}}
+				{{.ImportPath}}
+			    {{end}}' \
+			"${BUILDFLAGS[@]}" $TEST_PATH \
+			| grep github.com/docker/docker \
+			| grep -v github.com/docker/docker/vendor \
+			| grep -v github.com/docker/docker/daemon/graphdriver \
```

Comment:
@LK4D4: My only intention there was to get daemon/graphdriver tests to not run for Solaris. We support only zfs and running the other graph plugins doesn't make sense.
With my limited understanding of hack/make/test-unit this seemed to be the simplest way to do it.
Any logic that would help accomplish this better is fine.